packages/graphql/lib/src/core/observable_query.dart

```diff
@@ -251,6 +251,7 @@ class ObservableQuery<TParsed> {
   void addResult(QueryResult<TParsed> result, {bool fromRebroadcast = false}) {
     // don't overwrite results due to some async/optimism issue
     if (latestResult != null &&
+        latestResult!.source == QueryResultSource.network &&
         latestResult!.timestamp.isAfter(result.timestamp)) {
       return;
```

Comment:
```suggestion
    if (latestResult?.source == QueryResultSource.network &&
        latestResult?.timestamp.isAfter(result.timestamp)) {
      return;
```

Last time I was looking to our code and I was quite scary  on how we hard relay on the if else case, this imho make the code quite unreliable  and difficult to debug.

I think this is an equivalent dart if? I'm missing something on it?

Reply:
Yeah, I just tried to respect old code convention.
But I think your change would work with dart dart optional chaining safely well and seems better as you told.
Except, Line 254; `latestResult?.timestamp.isAfter(result.timestamp)` can evaluate `null` not `boolean` so might need to be `latestResult?.timestamp.isAfter(result.timestamp) == true`.

Reply:
applied